### PR TITLE
Embedded retro dev

### PR DESCRIPTION
**Pull request by ezraboley**
# Description

Please include a summary of the change and which issue (if applicable) is fixed

All about those retros and a GPIO C interface. There is no voting yet, but I thought we should get all these changes into master and then have a Nav branch that could work out the particulars of that issue

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

# Checklist

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (BIG TIME)
- [X] My changes generate no new warnings or errors
- [X] I have added tests/examples that prove my fix is effective or that my feature works
- [X] New and existing tests/examples pass with my changes

### CODE REVIEW

**Comment by ezraboley** [comment]
Ok the only reason it fails Travis is that Travis hates the color flag on GCC. I actually really like the color flag so I take issue with that and I'll see if Travis can be forced to use a newer GCC

**Comment by ezraboley** [comment]
Yay it passes Travis again (apparently it's surprising difficult to get it not to use a super out of date version of GCC by default)

**Comment by EUdds** [comment]
Looks good

**Review by EUdds — DISMISSED** [review]
Looks good!

File: embedded/app/src/states.c [hunk 1]
```diff
@@ -37,7 +37,7 @@ extern stateTransition_t *findTransition(state_t *currState, char *name);
  */
 
 static bool checkStopped(void) {
-	return fabs(data->motion->vel) < MAX_STOPPED_VEL &&  (time(NULL) - data->timers->lastRetro) > 15;
+	return fabs(data->motion->vel) < MAX_STOPPED_VEL &&  (getuSTimestamp() - data->timers->lastRetro) > 15000000;
```
Comment (rdaruwala) [inline embedded/app/src/states.c]: Can we make the 15/15000000 a const and not a floating number? (This is my err from earlier but we can get it fixed now)

File: embedded/data/include/data.h [hunk 2]
```diff
@@ -41,12 +61,10 @@ typedef struct data_t {
...
-    time_t startTime;
-    time_t lastRetro;
-    time_t lastRetro1;
-    time_t lastRetro2;
-    time_t lastRetro3;
-    time_t timeInState;
+    uint64_t startTime;
+    uint64_t lastRetro;
+	uint64_t lastRetros[NUM_RETROS];
+    uint64_t timeInState;
```
Comment (rdaruwala) [inline embedded/data/include/data.h]: Small note -- Do we actually use `timeInState` anywhere? Do we need it?

**Comment by rdaruwala** [comment]
Made 2 quick comments -- Nothing significant but would be 👍 if we could address before merging (so we don't have to deal with it in the next PR)

**Review by EUdds — APPROVED** [review]
Looks good... again!

**Pull request merged**